program-runtime/src/execution_budget.rs

```diff
@@ -0,0 +1,252 @@
+            sysvar_base_cost: 100,
+            secp256k1_recover_cost: 25_000,
+            syscall_base_cost: 100,
+            curve25519_edwards_validate_point_cost: 159,
+            curve25519_edwards_add_cost: 473,
+            curve25519_edwards_subtract_cost: 475,
+            curve25519_edwards_multiply_cost: 2_177,
+            curve25519_edwards_msm_base_cost: 2_273,
+            curve25519_edwards_msm_incremental_cost: 758,
+            curve25519_ristretto_validate_point_cost: 169,
+            curve25519_ristretto_add_cost: 521,
+            curve25519_ristretto_subtract_cost: 519,
+            curve25519_ristretto_multiply_cost: 2_208,
+            curve25519_ristretto_msm_base_cost: 2303,
+            curve25519_ristretto_msm_incremental_cost: 788,
+            heap_size: u32::try_from(solana_program_entrypoint::HEAP_LENGTH).unwrap(),
+            heap_cost: DEFAULT_HEAP_COST,
+            mem_op_base_cost: 10,
+            alt_bn128_addition_cost: 334,
+            alt_bn128_multiplication_cost: 3_840,
+            alt_bn128_pairing_one_pair_cost_first: 36_364,
+            alt_bn128_pairing_one_pair_cost_other: 12_121,
+            big_modular_exponentiation_base_cost: 190,
+            big_modular_exponentiation_cost_divisor: 2,
+            poseidon_cost_coefficient_a: 61,
+            poseidon_cost_coefficient_c: 542,
+            get_remaining_compute_units_cost: 100,
+            alt_bn128_g1_compress: 30,
+            alt_bn128_g1_decompress: 398,
+            alt_bn128_g2_compress: 86,
+            alt_bn128_g2_decompress: 13610,
+        }
+    }
+
+    /// Returns cost of the Poseidon hash function for the given number of
+    /// inputs is determined by the following quadratic function:
+    ///
+    /// 61*n^2 + 542
+    ///
+    /// Which aproximates the results of benchmarks of light-posiedon
+    /// library[0]. These results assume 1 CU per 33 ns. Examples:
+    ///
+    /// * 1 input
+    ///   * light-poseidon benchmark: `18,303 / 33 ≈ 555`
+    ///   * function: `61*1^2 + 542 = 603`
+    /// * 2 inputs
+    ///   * light-poseidon benchmark: `25,866 / 33 ≈ 784`
+    ///   * function: `61*2^2 + 542 = 786`
+    /// * 3 inputs
+    ///   * light-poseidon benchmark: `37,549 / 33 ≈ 1,138`
+    ///   * function; `61*3^2 + 542 = 1091`
+    ///
+    /// [0] https://github.com/Lightprotocol/light-poseidon#performance
+    pub fn poseidon_cost(&self, nr_inputs: u64) -> Option<u64> {
+        let squared_inputs = nr_inputs.checked_pow(2)?;
+        let mul_result = self
+            .poseidon_cost_coefficient_a
+            .checked_mul(squared_inputs)?;
+        let final_result = mul_result.checked_add(self.poseidon_cost_coefficient_c)?;
+
+        Some(final_result)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct SVMTransactionComputeBudgetAndLimits {
+    pub budget: SVMTransactionExecutionBudget,
+    pub loaded_accounts_bytes: NonZeroU32,
+    pub priority_fee: u64,
+}
```

Comment:
Are we going to keep both `SVMTransactionComputeBudgetAndLimits` and `ComputeBudgetLimits`? I see you've also declared `ComputeBudget` as an alias form `SVMTransactionExecutionBudget`, so I believe the usage and semantics got a little confusing.

Reply:
`ComputeBudgetLimits` will stay in monorepo. It has more information than needed by SVM. `SVMTransactionComputeBudgetAndLimits` contains all that's needed to process a transaction by SVM. So that'll be the interface SVM will export.